ngraph/frontend/frontend_manager/include/frontend_manager/frontend_manager.hpp

```diff
@@ -0,0 +1,518 @@
+            virtual void cut_and_add_new_input(Place::Ptr place,
+                                               const std::string& newNameOptional = "");
+
+            /// \brief Cut immediately after this place and assign this place as new output; prune
+            /// all nodes that don't contribute to any output.
+            /// \param place New place to be assigned as output
+            /// \param newNameOptional Optional new name assigned to this output place
+            virtual void cut_and_add_new_output(Place::Ptr place,
+                                                const std::string& newNameOptional = "");
+
+            /// \brief Assign this place as new output or add necessary nodes to represent a new
+            /// output.
+            ///
+            /// \param place Anchor point to add an output
+            /// \return new output place, may be the same as a given place
+            virtual Place::Ptr add_output(Place::Ptr place);
+
+            /// \brief Removes any sinks directly attached to this place with all inbound data flow
+            /// if it is not required by any other output.
+            /// \param place Model place
+            virtual void remove_output(Place::Ptr place);
+
+            /// \brief Replaces all existing outputs with new ones removing all data flow that is
+            /// not required for new outputs.
+            ///
+            /// \param outputs Vector with places that will become new outputs; may intersect
+            /// existing outputs.
+            /// \param outputs Array of new output places
+            virtual void override_all_outputs(const std::vector<Place::Ptr>& outputs);
+
+            /// \brief Modifies the graph to use new inputs instead of existing ones. New inputs
+            /// should completely satisfy all existing outputs.
+            /// \param inputs Array of new input places
+            virtual void override_all_inputs(const std::vector<Place::Ptr>& inputs);
+
+            /// \brief Leaves only subgraph that are defined by new inputs and new outputs.
+            /// \param inputs Array of new input places
+            /// \param outputs Array of new output places
+            virtual void extract_subgraph(const std::vector<Place::Ptr>& inputs,
+                                          const std::vector<Place::Ptr>& outputs);
+
+            ///// Setting tensor properties  /////
+
+            /// \brief Defines all possible shape that may be used for this place; place should be
+            /// uniquely refer to some data. This partial shape will be converted to corresponding
+            /// shape of results ngraph nodes and will define shape inference when the model is
+            /// converted to ngraph.
+            /// \param place Model place
+            /// \param shape Partial shape for this place
+            virtual void set_partial_shape(Place::Ptr place, const ngraph::PartialShape& shape);
```

Comment:
Why we have specialized methods for Place inside InputModel?

Reply:
Per my understanding, this is because in general Place may not know it's partial shape, element type, tensor values. So, the actual model is responsible for setting these properties for Place. @slyalin is my understanding correct?

Reply:
Yes. Places don't act as model modifiers. Having just a place you can only read model properties, but don't modify. It is a kind or read/write access rights separation. InputModel is only object that accepts modification requests, Place is only for pointing to various... places in a model.

Reply:
Marking as resolved